src/logging.rs

```diff
@@ -0,0 +1,61 @@
+use std::io::Write;
+
+pub(crate) struct PgxGuestWriter<const IS_STDERR: bool>;
+
+impl<const IS_STDERR: bool> Write for PgxGuestWriter<IS_STDERR> {
+    fn write(&mut self, data: &[u8]) -> std::result::Result<usize, std::io::Error> {
+        let content = std::str::from_utf8(data).expect("Could not interpret stdout as UTF-8");
```

Comment:
Hmm. I am not following how non-UTF8 data would get into our stdout/stderr yet...? Is there no way we can ban that on the type level?

Reply:
Apparently "theoretically a build log could". Eh, I suppose.

Reply:
In theory a build.rs script error log could dumb non-UTF-8 data to stdout but that's highly irregular.